virttest/qemu_devices/qcontainer.py

```diff
@@ -597,7 +597,8 @@ def simple_hotplug(self, device, monitor, bus=None):
         """
         self.set_dirty()
 
-        if isinstance(device, qdevices.QDevice):
+        if isinstance(device, qdevices.QDevice) and \
+                "bus" in self.execute_qemu("-device %s,?" % device["driver"]):
```

Comment:
That's not safe, actually most QDevice has `bus` option does not include `bus` in its `-device %s,?`, e.g. `nec-usb-xhci`, `virtio-pci` has `bus` but not exactly option `bus`, it just luckily has an option `virtio-pci-bus-master-bug-migration=bool (on/off)`

Reply:
It looks like `/usr/libexec/qemu-kvm -device ?` will give exactly the device has bus or not.

Reply:
@vivianQizhu  I thought `-device ?` would show all the options it accepts, but apparently I was wrong.  I checked everywhere but didn't find a way to list all the available options.  Seems this approach is not workable.

Reply:
@huangyum `-device ?` actually shows the bus of device, e.g.:
```
Input devices:
name "ccid-card-passthru", bus ccid-bus, desc "passthrough smartcard"
name "isa-serial", bus ISA
name "pci-serial", bus PCI
```
```
Uncategorized devices:
name "edu", bus PCI
name "intel-iommu", bus System
name "nvdimm", desc "DIMM memory module"
name "pc-dimm", desc "DIMM memory module"
name "virtio-gpu-device", bus virtio-bus
name "vmgenid"
```
Exactly shows the device has bus or not, and what is the bus type of the device.